### PR TITLE
Fix CodeQL Findings 2022-12-23

### LINKED ISSUE #343: Superclass attribute shadows subclass method in smtp.py
This is a valid issue. Gotta rename the private method to something that's not shadowing.

Affected file: **`smtp.py`**

Tracking issue for:
- [ ] https://github.com/aio-libs/aiosmtpd/security/code-scanning/5

### PR DESCRIPTION
**Pull request by pepoluan**
## What do these changes do?

Fix some more significant CodeQL warnings, i.e., those that can't be dismissed outright.

## Are there changes in behavior for the user?

None

## Related issue number

Closes #343 , #344 , #345 , #346, #349, #350, #351

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist

### CODE REVIEW

**Comment by pepoluan** [comment]
Don't you hate intermittent test failures... I'm barging on because the test failures are _really_ intermittent. I mean, like one OS with one particular Python version fails, while ALL others succeeds. Then the next iteration a different combination fails... with no reason nor logic.

Luckily our GHCI workflows have been designed so it would be an easy -- and quite fast -- task to just rerun the failed tests. But I'm delaying that until this series of commits and pushes are done.

File: aiosmtpd/proxy_protocol.py [hunk 1]
```diff
@@ -102,17 +102,9 @@
...
-        return b""
-
-    async def readexactly(self, num_bytes: int) -> bytes:
-        ...
-        return b""
-
-    async def readuntil(self, until_chars: Optional[bytes] = None) -> bytes:
-        ...
-        return b""
+    async def read(self, num_bytes: Optional[int] = None) -> bytes: ...
```
[the remark this answers is not in the document]
Comment (pepoluan) [reply aiosmtpd/proxy_protocol.py]: IT'S FRIKKING PROTOCOLS DEFINITION

File: aiosmtpd/proxy_protocol.py [hunk 2]
```diff
@@ -102,17 +102,9 @@
...
+    async def readexactly(self, num_bytes: int) -> bytes: ...
```
[the remark this answers is not in the document]
Comment (pepoluan) [reply aiosmtpd/proxy_protocol.py]: IT'S FRIKKING PROTOCOLS DEFINITION

File: aiosmtpd/proxy_protocol.py [hunk 3]
```diff
@@ -102,17 +102,9 @@
...
+    async def readuntil(self, until_chars: Optional[bytes] = None) -> bytes: ...
```
[the remark this answers is not in the document]
Comment (pepoluan) [reply aiosmtpd/proxy_protocol.py]: IT'S FRIKKING PROTOCOLS DEFINITION

**Comment by pepoluan** [comment]
When merged, this PR fixes all outstanding CodeQL warnings (those that I haven't dismissed outright) [except one](https://github.com/aio-libs/aiosmtpd/security/code-scanning/10). That one requires some deeper thought as it's a decorator and things get reeeeeallly hairy there...

No real urgency, as without the fixes `aiosmtpd` actually Just Works. But it's good to see no outstanding "security" item in the Security tab 😄

**Comment by pepoluan** [comment]
Adding @warsaw here because there are changes that applies to his code 😉

**Comment by warsaw** [comment]
Apologies for the delayed response.  Thanks for merging this one!

**Pull request merged**